`Hds::SideNav` - add `@isCollapsable` and `@isMinimized` arguments (#1630)
hashicorp/design-system: website/docs/components/side-nav/partials/code/component-api.md

  <C.Property @name="isResponsive" @type="boolean" @default="true">
    Controls whether the SideNav is responsive to viewport changes. It can be programmatically turned off by passing `false`.
    <br>

>     <em>Notice: even if the `@isResponsive` parameter is set to false, some JavaScript is executed anyway in the background, and event listeners are attached to some DOM elements (even if this functionality is not used).</em>

Reviewer: ```suggestion
    <em>Notice: even if the `@isResponsive` parameter is set to false, some JavaScript is still executed in the background, and event listeners are attached to some DOM elements (even if this functionality is not used).</em>
```

  </C.Property>
  <C.Property @name="isCollapsible" @type="boolean" @default="false">

>     Controls whether the SideNav is collapsible on large viewports. When this argument and `isResponsive` are set to `true` a toggle button will permanently be rendered to collapse and expand the SideNav.

Reviewer 2: ```suggestion
    Controls whether the SideNav can be collapsed by the user on large viewports. When this argument and `isResponsive` are set to `true` a toggle button will permanently be rendered to collapse and expand the SideNav.
```

>     Controls whether the SideNav is collapsible on large viewports. When this argument and `isResponsive` are set to `true` a toggle button will permanently be rendered to collapse and expand the SideNav.

Reviewer: ```suggestion
    Controls whether the SideNav is collapsible on large viewports. When this argument and `isResponsive` are set to `true`, a toggle button will permanently be rendered to collapse and expand the SideNav.
```

    <br>

>     <em>Notice: if `@isResponsive` is set to false this argument has no effects.</em>

Reviewer: ```suggestion
    <em>Notice: if `@isResponsive` is set to false, this argument has no effect.</em>
```

  </C.Property>
  <C.Property @name="isMinimized" @type="boolean" @default="false">

>     Controls if the `SideNav` is rendered collapsed or expanded when initialized. This allows an application to preserve the collapsed/expanded state across sessions.

Reviewer 2: Maybe rewrite my suggestion, I couldn't come up with a clearer formulation of what I intended to say :)

>     Controls if the `SideNav` is rendered collapsed or expanded when initialized. This allows an application to preserve the collapsed/expanded state across sessions.

Reviewer 3: ```suggestion
    Controls if the SideNav is rendered collapsed or expanded when initialized. This allows an application to preserve the collapsed/expanded state across sessions.
```

Removed backticks for consistency
